File: 82-remove-duplicates-from-sorted-list-ii/82-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        sanitel = ListNode(-101)
        sanitel.next = head
        p = sanitel.next
        q = sanitel
        while(p):
            r = p
            while(r.next and r.val == r.next.val):
                r = r.next
            if r == p:
                q.next = p
                q = q.next
                p = p.next
            else:
                p = r.next
        q.next = p
        return sanitel.next
        
            
            
            
        

File: 82-remove-duplicates-from-sorted-list-ii/82-remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dumb = new ListNode();
        dumb.next = head;
        ListNode prev = dumb;
        ListNode cur = head;
        
        while(cur != null){
            ListNode nxt = cur;
            int countDuplicate = 0;
            ListNode curPrev = prev;
            while(nxt != null && nxt.val == cur.val){
                curPrev = nxt;
                nxt = nxt.next;
                countDuplicate++;
            }
            cur = nxt;
            if(countDuplicate > 1){
                curPrev.next = null;
                prev.next = cur;
            }else{
                prev = curPrev;
            }
        }
        return dumb.next;
        
    }
}